### ISSUE TITLE
Eka's plugin doesn't work on some individual submission pages

### ISSUE THREAD

**Issue by Eupeptic1** [Simon-Tesla/RaccoonyWebEx#41]
**What is broken:**
Some submission pages at Eka's now have a box under the main submission and above the submission info (title, description, tags, etc), with thumbnails of other submissions in that same folder.

If this box is present, the Eka's plugin doesn't work on that particular submission page - you never get the raccoon-head icon.

Not all submission pages have this new box.  I'm not sure whether the artist can choose to display it, or if it appears automatically for submissions after a certain date.

For submission pages without this new box, the Eka's plugin works as it always did.

**What didn't break:**
"Open all in tabs" from a gallery page still works.

If you get the raccoon head on a submission page, everything you can do from its menu still works (download submission, launch OS file manager in downloads folder after download, full-screen view of submission, open settings screen)

**Steps to reproduce:**
To see the bug, visit one of these pages.  They both have the new thumbnail box.  The raccoon head will not appear - the expected result is that it does appear.

https://aryion.com/g4/view/531995 (character reference, more or less sfw)
https://aryion.com/g4/view/552116 (violence, nsfw)

To not see the bug (as a test), visit this page, which does not have the new thumbnail box.  The raccoon head will appear, which is the expected result.

https://aryion.com/g4/view/523092 (soft, probably nsfw)

**Environment**
Raccoony WebExtension 1.2.2
Firefox 60.9.0esr, 64-bit, desktop
Debian Linux 9.9, 64-bit

**Comment by Eupeptic1** [comment]
**Initial analysis**
The markup structure of pages with the new thumbnail box has changed, and because of that, the Eka's plugin fails to find the description.  When the parsing fails, it creates an error that stops the raccoon head from displaying.

The problem comes when the plugin is looking for the artist's description of the submission.  The page element that contains that description does not have a unique identifier.  At the time the plugin was written, there were at least three <div class="g-box-contents"> on the page, and the description was inside the second one; the plugin just always picked the second div of that class.  Currently, on a page without the thumbnail box, this same structure applies, so picking the second div of that class still works.

On a page with the thumbnail box, the thumbnail box itself is also inside a <div class="g-box-contents">, which is between the previous two divs of that class.  The plugin *should* grab the third div of that class, but it's hard-coded to grab the second.  The plugin then proceeds to look for a <p> inside that div to find the description text, but since it's got the wrong div, there isn't any, which results in an error.

I'm not 100% sure if the thumbnail box feature existed on the site back in fall 2017, when I initially wrote the plugin, and I just missed it, or whether it's a change at the site since then.  Either way, it still breaks on some pages now.

**Probable solution**
Find some other way to determine where the description is.  The description is not available anywhere else on the page (such as a meta tag); it's only in an un-named <p> inside *one* of the <div class="g-box-contents"> on the page.

Counting up from the bottom of the page probably won't work, because the comments (if any) also live in a <div class="g-box-contents">, so there may be a variable number of them.

Probably I need to git gud at querySelector().

**Comment by Eupeptic1** [comment]
I am testing a possible fix locally.  Basically, I grab the second div and see if it's a thumbnail box or not.  If it isn't, I can get the description from it.  If it is, I move to the third div, and get the description from that.

More to come in a day or two.

**Comment by Eupeptic1** [comment]
I've tested my change and it seems to be working.  I tried it on the examples I linked above and it works correctly; I also tried it on a few more existing submissions at Eka's I already had bookmarked.  I also logged into Eka's and tried it on about 30 new submissions that have the new thumbnail box, and a further 30 that don't, and it did the right thing in all of those cases.

Now, to fight with Git.
